ixmp4/data/db/base.py

```diff
@@ -391,7 +391,10 @@ def bulk_upsert_chunk(self, df: pd.DataFrame) -> None:
             for col in self.model_class.updateable_columns:
                 updated_col = col + self.merge_suffix
                 if updated_col in df.columns:
-                    cond.append(df[col] != df[updated_col])
+                    df[updated_col] = df[updated_col].astype(df[col].dtype)
```

Comment:
Maybe add a line explaining why this is necessary? And a test (in a GitHub Action that explicitly installs pyarrow) to guard against regression?